### PR TITLE
I18N-1322 - Generate & document OpenAPI Specification on Mojito APIs + Swagger UI

### PR DESCRIPTION
**Pull request by DarKhaos**
Generated and documented Open API Specification

### CODE REVIEW

File: webapp/src/main/java/com/box/l10n/mojito/entity/ApplicationCacheType.java [hunk 1]
```diff
@@ -27,6 +27,7 @@ public class ApplicationCacheType {
   @JsonView(View.IdAndName.class)
   protected short id;
 
+  @JsonView(View.Default.class)
```
Comment (DarKhaos) [inline webapp/src/main/java/com/box/l10n/mojito/entity/ApplicationCacheType.java:30]: Fields without @JsonView annotation were included in the response of an annotated endpoint that uses an specific view

File: webapp/src/main/java/com/box/l10n/mojito/entity/AssetExtraction.java [hunk 2]
```diff
@@ -46,6 +47,7 @@
 public class AssetExtraction extends AuditableEntity {
 
   @JsonBackReference("asset")
+  @Schema(hidden = true)
```
Comment (DarKhaos) [inline webapp/src/main/java/com/box/l10n/mojito/entity/AssetExtraction.java:50]: @JsonBackReference is not supported by Swagger, and these fields were included in the schema response

File: webapp/pom.xml [hunk 3]
```diff
@@ -140,6 +140,12 @@
             <artifactId>micrometer-registry-statsd</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+            <version>2.6.0</version>
```
Comment (maallen) [inline webapp/pom.xml:146]: Extract the version value into a maven property and then reference that property here, makes it straightforward for version updates in future.

File: webapp/src/main/java/com/box/l10n/mojito/rest/View.java [hunk 4]
```diff
@@ -7,6 +7,8 @@
  * @author jyi
  */
 public class View {
+  public interface Default {}
```
Comment (maallen) [inline webapp/src/main/java/com/box/l10n/mojito/rest/View.java:10]: Can you explain why this is needed? I expect it's related to @JSONVIEW but I'm not clear on the problem

File: webapp/src/main/java/com/box/l10n/mojito/entity/AssetExtraction.java (hunk 2 above)
Comment (DarKhaos) [reply webapp/src/main/java/com/box/l10n/mojito/entity/AssetExtraction.java:50]: it is not displayed in the HTTP response, but it shows up in the response schema

File: webapp/src/main/java/com/box/l10n/mojito/security/WebSecurityConfig.java [hunk 5]
```diff
@@ -173,7 +173,8 @@ public SecurityFilterChain configure(HttpSecurity http) throws Exception {
                     "/cli/**",
                     "/js/**",
                     "/css/**",
-                    "/error")
+                    "/error",
+                    "/v3/api-docs")
```
[the remark this answers is not in the document]
Comment (DarKhaos) [reply webapp/src/main/java/com/box/l10n/mojito/security/WebSecurityConfig.java]: Done

**Review by maallen — APPROVED** [review]

**Review by byronantak — APPROVED** [review]

File: webapp/src/main/java/com/box/l10n/mojito/Application.java [hunk 6]
```diff
@@ -50,6 +52,9 @@ public class Application {
   boolean shouldIndentJacksonOutput;
 
   public static void main(String[] args) throws IOException {
+    // Added a custom converter to hide fields that are not annotated with the @JsonView annotation
+    // in Swagger
+    ModelConverters.getInstance().addConverter(new JsonViewAwareModelResolver(Json.mapper()));
```
[the remark this answers is not in the document]
Comment (DarKhaos) [reply webapp/src/main/java/com/box/l10n/mojito/Application.java:57]: Yes, it only hides fields from the Swagger schemas

**Pull request merged**